frontend/online-shop/src/components/productDetails/ProductDetails.tsx

```diff
@@ -0,0 +1,54 @@
+import { useStyle } from "../../styles/productDetails/ProductDetailsStyle";
+import { Grid, IconButton } from "@material-ui/core";
+import ShoppingCart from "@material-ui/icons/ShoppingCart";
+import React from "react";
+
+export const ProductDetails = (props: any) => {
+  const style = useStyle();
+
+  const { number } = props.match.params;
+  const { product } = props.location.state;
```

Comment:
As this is a component that is directly tied to a route you shouldn't use props with it. Indeed when you come to this page from the products list page you will be able to send the product and retrieve it from the props. How about when you try to access directly `http://localhost:4200/products/1`? Also there is a method called useParams() that can retrieve the path parameters object without using props